client/src/components/RatingTile.js

```diff
@@ -0,0 +1,98 @@
+import React, { useState } from "react";
+
+const RatingTile = ({ id, content, activityId }) => {
+  const [ratings, setRatings] = useState({});
+  const [allRatings, setAllRatings] = useState([]);
+
+  const postRating = async (reviewId, selectedRating) => {
+    try {
+      const response = await fetch(`/api/v1/activities/${activityId}/reviews/rating`, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({ reviewId, rating: selectedRating }),
+      });
+      if (!response.ok) {
+        throw new Error(`${response.status} (${response.statusText})`);
+      }
+      const responseData = await response.json();
+      setRatings(responseData);
+    } catch (error) {
+      console.error("Error in fetch!", error.message);
+    }
+  };
+
+  let upVotes = 0;
+  let downVotes = 0;
+
+  allRatings.forEach((ratingObject) => {
+    if (id === ratingObject.reviewId && ratingObject.rating === 2) {
+      upVotes++;
+    }
+    if (id === ratingObject.reviewId && ratingObject.rating === 1) {
+      downVotes++;
+    }
+  });
+  const totalVotes = upVotes - downVotes;
+
+  const getRatings = async () => {
+    try {
+      const response = await fetch(`/api/v1/activities/${activityId}/reviews/rating`);
+      if (!response.ok) {
+        const error = new Error(`${response.status} (${response.statusText})`);
+        throw error;
+      }
+      const responseData = await response.json();
+      setAllRatings(responseData.allRatings);
+    } catch (error) {
+      console.error("Error in fetch!", error.message);
+    }
+  };
+
+  const handleRating = ({ reviewId, selectedRating }) => {
+    postRating(reviewId, selectedRating);
+    getRatings();
+  };
+
+  if (ratings.newRating) {
+    return (
+      <li>
+        {content}
+        <div className="voting-div thumbs-down">{ratings.newRating}</div>
+        <div className="thumbs-up">{`Likes: ${upVotes}`}</div>
+        <div className="thumbs-down">{`Dislikes: ${downVotes}`}</div>
+        <div className="">{`Total: ${totalVotes}`}</div>
+      </li>
+    );
```

Comment:
first
- this could be a component!
- the only difference between 62-65 and 88-94 is one line. We can use some simpler, nested logic to dry this up